cdap-ui/app/cdap/components/DataPrep/TopPanel/index.js

```diff
@@ -88,6 +94,56 @@ export default class DataPrepTopPanel extends Component {
     );
   }
 
+  onSubmit() {
+    if (this.props.onSubmit) {
+      let directives = DataPrepStore.getState().dataprep.directives;
+      let workspaceId = DataPrepStore.getState().dataprep.workspaceId;
```

Comment:
Is the content of this function the same as the one in AddToPipeline implementation? If it is, can we extract it to a helper function?